File: cspro-dev/cspro/CSEntryDroid/app/src/main/java/gov/census/cspro/form/CaseTreeUpdate.kt
package gov.census.cspro.form

data class CaseTreeUpdate(
    val type: Int,
    val node: CaseTreeNode?,
    val parentNodeId: Int,
    val childIndex: Int) {

    companion object {
        const val NODE_MODIFIED = 1
        const val NODE_ADDED = 2
        const val NODE_REMOVED = 3
    }
}

File: cspro-dev/cspro/Zentryo/CaseTreeUpdate.cpp
#include "StdAfx.h"
#include "CaseTreeUpdate.h"

CaseTreeUpdate::CaseTreeUpdate(UpdateType type, std::shared_ptr<CaseTreeNode> node, int childIndex)
    : m_type(type),
      m_node(node),
      m_childIndex(childIndex)
{
}

CaseTreeUpdate CaseTreeUpdate::CreateModify(std::shared_ptr<CaseTreeNode> node)
{
    return CaseTreeUpdate(UpdateType::NODE_MODIFIED, node, -1);
}

CaseTreeUpdate CaseTreeUpdate::CreateAdd(std::shared_ptr<CaseTreeNode> parentNode, int childIndex)
{
    return CaseTreeUpdate(UpdateType::NODE_ADDED, parentNode, childIndex);
}

CaseTreeUpdate CaseTreeUpdate::CreateRemove(std::shared_ptr<CaseTreeNode> parentNode, int childIndex)
{
    return CaseTreeUpdate(UpdateType::NODE_REMOVED, parentNode, childIndex);
}
